### PR TITLE
Add support for overriding taskToDomain in a SUB_WORKFLOW system task

### PR DESCRIPTION
**Pull request by xesxen**

### CODE REVIEW

File: core/src/main/java/com/netflix/conductor/core/execution/mapper/SubWorkflowTaskMapper.java [hunk 1]
```diff
@@ -27,6 +27,8 @@
 
 import javax.inject.Inject;
 import java.util.Collections;
+import javax.swing.text.html.Option;
```
Comment (apanicker-nflx) [inline core/src/main/java/com/netflix/conductor/core/execution/mapper/SubWorkflowTaskMapper.java]: please remove this unused dependency

**Comment by xesxen** [comment]
Hi @apanicker-nflx, I added several tests. Please review the patch again!

**Comment by apanicker-nflx** [comment]
@xesxen Thanks for adding the tests. The unit tests look good and cover most of the changes.

Please also add integration tests to the `test-harness` module in the `https://github.com/Netflix/conductor/blob/master/test-harness/src/test/java/com/netflix/conductor/tests/integration/AbstractWorkflowServiceTest.java` class

**Comment by xesxen** [comment]
@apanicker-nflx Can you review the PR again?

**Comment by apanicker-nflx** [comment]
@xesxen Thanks for the contribution. PR looks good.

File: common/src/test/java/com/netflix/conductor/common/workflow/SubWorkflowParamsTest.java [hunk 2]
```diff
@@ -7,9 +7,7 @@
 import javax.validation.Validation;
 import javax.validation.Validator;
 import javax.validation.ValidatorFactory;
-import java.util.ArrayList;
-import java.util.List;
-import java.util.Set;
+import java.util.*;
```
Comment (demichej) [inline common/src/test/java/com/netflix/conductor/common/workflow/SubWorkflowParamsTest.java]: No idea if you'll get this, but you shouldnt actually use wildcards in imports, should be explicit.

**Pull request merged**